### DISCUSSION TITLE
关于自定义扩展组件的问题

### DISCUSSION THREAD

**Discussion by lateautumn2** [YunYouJun/valaxy#141] [Q&A]
你好,云游君,关于自定义组件,我有一点疑问,比如我需要在博客左下角摆放一个盒子,为此我新建了一个vue组件,但是我要怎么使用,才能让这个组件变成默认的布局 **(不使用自定义同名组件覆盖,这个方法我试过没有问题,也不在markdown中使用layout布局,这个也不算是默认布局了,而是文档布局)**,我在valaxy文档以及你的domo中并没有找到使用方法,目前文档中的方法,我只看到覆盖使用方法和markdown中使用的方法

**Comment by YunYouJun** [comment]
在 `layouts` 文件夹下新建 `xxx.vue`（你的自定义布局名） 即可。

可参考 https://github.com/YunYouJun/valaxy/blob/main/demo/yun/layouts/docs.vue

**Reply by lateautumn2** [reply]
自定义组件不再支持了吗，目前我无法往里面插入任何东西

**Reply by YunYouJun** [reply]
支持的，你指的用法是？

**Reply by lateautumn2** [reply]
在 layouts 文件夹下新建 xxx.vue（你的自定义布局名） 即可。

可参考 https://github.com/YunYouJun/valaxy/blob/main/demo/yun/layouts/docs.vue

--------------------------------------------------------------------------------
以上这个，我没理解错的话，是以插槽的形式来使用的吧？

**Reply by lateautumn2** [reply]
我看了你上面给的示例，是在文章中使用layout新的布局，不是我想要的结果
例：我想在banner页的上面再添加一个组件放我需要的信息，但是我想不修改YunBanner(插槽或者覆盖)，我想直接在layouts或者components中新建一个vue文件，然后能否在何处引用，直接append到banner上面，我需要怎么做

**Reply by YunYouJun** [reply]
新建 `layouts/home.vue`（覆盖主题的 layouts/home.vue 布局），复制 [valaxy-theme-yun/layouts/home.vue](https://github.com/YunYouJun/valaxy/blob/main/packages/valaxy-theme-yun/layouts/home.vue)，基于此进行修改，插入想要的内容。

![image](https://github.com/YunYouJun/valaxy/assets/25154432/500d3eed-f29b-41ed-9c15-6c2c35a4e040)

**Comment by lateautumn2** [comment]
question：我能否在components中编写的vue文件中调用valaxy.config.ts中defineValaxyConfig里面设置的值
target：我想要在自定义组件中重写默认的YunComponent，所以我需要拿到我在defineValaxyConfig设定的值，有什么方法可以直接拿出来吗

**Reply by YunYouJun** [reply]
```ts
import { useValaxyConfig } from 'valaxy'
const vConfig = useValaxyConfig()
```

**Reply by lateautumn2** [reply]
好的喵，今晚就回去试试
